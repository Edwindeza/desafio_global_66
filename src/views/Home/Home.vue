<style src="./style.scss" lang="scss" scoped></style>
<template src="./template.html"></template>

<script>
// @ is an alias to /src

import { mapActions } from 'vuex';

export default {
  name: 'Home',
  components: {},
  created() {
    setInterval(() => {
      this.hideLoader();
    }, 1000);
  },
  methods: {
    ...mapActions('listPokemons', ['hideLoader']),
  },
};
</script>
