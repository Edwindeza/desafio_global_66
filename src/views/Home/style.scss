.home{
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 100vh;
  &-content{
    max-width: 570px;
    h1{
      font-weight: bold;
      font-size: 26px;
      line-height: 31px;
      text-align: center;
      color: #353535;
      margin-bottom: 0;
      padding-top: 78px;
      padding-bottom: 30px;
    }
    p{
      font-weight: 500;
      font-size: 18px;
      line-height: 150%;
      text-align: center;
      color: #5E5E5E;
    }
    a{
      display: block;
      padding: 11px 20px;
      background: #F22539;
      border-radius: 60px;
      color: white;
      width: fit-content;
      text-decoration: none;
      margin: auto;
    }
  }
}