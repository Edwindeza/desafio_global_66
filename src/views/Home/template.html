
  <div class="home">
    <div class="home-content">
      <img alt="Vue logo" src="@/assets/pikachu.svg">
      <h1>Welcome to Pokédex</h1>
      <p>
        The digital encyclopedia created by Professor Oak is an invaluable tool to Trainers in the Pokémon world.
      </p>
      <router-link to="/pokedex">Get started</router-link>
    </div>
    
  </div>