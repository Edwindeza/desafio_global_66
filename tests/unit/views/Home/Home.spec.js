import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import Home from '@/views/Home/Home.vue';

const wrapper = shallowMount(Home, {
  stubs: {
    RouterLinkStub,
  },
});

describe('Prueba Home Component', () => {
  test('Debe de hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe('Prueba elementos', () => {
  test('h1 debe coincidir con texto de prueba ', () => {
    const h1Value = 'Welcome to Pokédex';
    expect(wrapper.find('h1').text()).toBe(h1Value);
  });
  test('p debe coincidir con texto de prueba ', () => {
    const pValue = 'The digital encyclopedia created by Professor Oak is an invaluable tool to Trainers in the Pokémon world.';
    expect(wrapper.find('p').text()).toBe(pValue);
  });
  test('router-link debe coincidir con texto de prueba ', () => {
    const rlValue = 'Get started';
    expect(wrapper.find('router-link').text()).toBe(rlValue);
  });
});
