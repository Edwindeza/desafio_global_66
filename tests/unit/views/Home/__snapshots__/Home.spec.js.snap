// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba Home Component Debe de hacer match con el snapshot 1`] = `
<div class="home">
  <div class="home-content"><img alt="Vue logo" src="@/assets/pikachu.svg">
    <h1>Welcome to Pokédex</h1>
    <p>
      The digital encyclopedia created by Professor Oak is an invaluable tool to Trainers in the Pokémon world.
    </p>
    <router-link to="/pokedex">Get started</router-link>
  </div>
</div>
`;
